### PR TITLE
BFD-1938: restore db schema artifacts back to original names

### PR DESCRIPTION
**Pull request by mackec1**
**JIRA Ticket:**
[BFD-1938](https://jira.cms.gov/browse/BFD-1938)

**User Story or Bug Summary:**
Flyway migration script to drop current db views and rename __new_ tables by dropping the __new_ suffix; this results in table names that matched original table names prior to starting the schema migration work.

### What Does This PR Do?

- Drop all db views that point to __new_ tables.
- Rename all constraint and index names by removing any __new_ in their name(s).
- Rename all table names that have the suffix __new_ by dropping the suffix __new_.

### What Should Reviewers Watch For?

- both HSQL and PSQL (postgres) are correctly supported.
- All table, constraint, and index names adhere to a BFD naming convention.
- scripts are documented and well structured.

### What Security Implications Does This PR Have?

* Does this PR add any new software dependencies? 
  * [ ] Yes
  * [X] No
* Does this PR modify or invalidate any of our security controls?
  * [ ] Yes
  * [X] No
* Does this PR store or transmit data that was not stored or transmitted before?
  * [ ] Yes
  * [X] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [X] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* No prerequisites; **however**, it is expected that other flyway scripts may be merged into master prior to this script; this means that the flyway script name (versioning) will need need to be modified to ensure Flyway processing.

### Submitter Checklist

I have gone through and verified that...:

* [X] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [X] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [X] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [X] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [X] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [X] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [X] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V90__new_schema_rename_to_original.sql [hunk 1]
```diff
@@ -0,0 +1,534 @@
...
+    
+-- change the name of the index for mbi_hash
+alter index ${logic.psql-only} if exists 
+    public.beneficiaries_history_new_mbi_hash_idx rename to beneficiaries_history_mbi_hash_idx;
+
+-- for hsql, we previously dropped the primary key with cascade, so we need
+-- to actually create the fk constraint
+alter table if exists public.beneficiaries_history_new
+${logic.hsql-only} add constraint beneficiaries_history_to_beneficiaries foreign key (bene_id)
+${logic.hsql-only}     references public.beneficiaries_new (bene_id);
```
Comment (lsmitchell) [inline apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V90__new_schema_rename_to_original.sql:96]: will this reference be updated when we update the table name further down this script, or does this need to get updated again after the rename? (Or does this ref not matter after this line?)

File: apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V90__new_schema_rename_to_original.sql [hunk 2]
```diff
@@ -0,0 +1,534 @@
...
+-- =====================
+-- psql allows you to rename primary key
+${logic.psql-only} alter index if exists public.carrier_claim_lines_new_pkey
+${logic.psql-only}    rename to carrier_claim_lines_pkey;
+
+-- hsql doesn't support rename on primary key; we'll drop and re-create primary key
+${logic.hsql-only} alter table public.carrier_claim_lines_new
+${logic.hsql-only}     drop constraint public.carrier_claim_lines_new_pkey cascade;
+${logic.hsql-only} alter table public.carrier_claim_lines_new
+${logic.hsql-only}     add constraint carrier_claim_lines_new_pkey primary key (clm_id, line_num);
```
Comment (lsmitchell) [inline apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V90__new_schema_rename_to_original.sql:191]: this shouldnt have the _new right?

File: apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V90__new_schema_rename_to_original.sql [hunk 3]
```diff
@@ -0,0 +1,534 @@
...
+-- =====================
+-- psql allows you to rename primary key
+${logic.psql-only} alter index if exists public.dme_claim_lines_new_pkey
+${logic.psql-only}    rename to dme_claim_lines_pkey;
+
+-- hsql doesn't support rename on primary key; we'll drop and re-create primary key
+${logic.hsql-only} alter table public.dme_claim_lines_new
+${logic.hsql-only}     drop constraint public.dme_claim_lines_new_pkey;
+${logic.hsql-only} alter table public.dme_claim_lines_new
+${logic.hsql-only}     add constraint dme_claim_liness_new_pkey primary key (clm_id, line_num);
```
Comment (lsmitchell) [inline apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V90__new_schema_rename_to_original.sql:221]: think this also shouldnt have a _new, and has an extra 's'

File: apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V90__new_schema_rename_to_original.sql [hunk 4]
```diff
@@ -0,0 +1,534 @@
...
+-- =====================
+-- psql allows you to rename primary key
+${logic.psql-only} alter index if exists public.hha_claim_lines_new_pkey
+${logic.psql-only}    rename to hha_claim_lines_pkey;
+
+-- hsql doesn't support rename on primary key; we'll drop and re-create primary key
+${logic.hsql-only} alter table public.hha_claim_lines_new
+${logic.hsql-only}     drop constraint public.hha_claim_lines_new_pkey;
+${logic.hsql-only} alter table public.hha_claim_lines_new
+${logic.hsql-only}     add constraint hha_claim_lines_new_pkey primary key (clm_id, clm_line_num);
```
Comment (lsmitchell) [inline apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V90__new_schema_rename_to_original.sql:251]: Also has _new

File: apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V90__new_schema_rename_to_original.sql [hunk 5]
```diff
@@ -0,0 +1,534 @@
...
+-- =====================
+-- psql allows you to rename primary key
+${logic.psql-only} alter index if exists public.hospice_claim_lines_new_pkey
+${logic.psql-only}    rename to hospice_claim_lines_pkey;
+
+-- hsql doesn't support rename on primary key; we'll drop and re-create primary key
+${logic.hsql-only} alter table public.hospice_claim_lines_new
+${logic.hsql-only}     drop constraint public.hospice_claim_lines_new_pkey;
+${logic.hsql-only} alter table public.hospice_claim_lines_new
+${logic.hsql-only}     add constraint hospice_claim_lines_new_pkey primary key (clm_id, clm_line_num);
```
Comment (lsmitchell) [inline apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V90__new_schema_rename_to_original.sql:281]: (I guess just double check all of these, also has _new here and more below)

**Review by lsmitchell — CHANGES_REQUESTED** [review]

File: apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V90__new_schema_rename_to_original.sql (hunk 2 above)
Comment (mackec1) [reply apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V90__new_schema_rename_to_original.sql]: correct...yea there was a few of them. Eyes must have been glazing over...the good/bad news is that hsql still worked, even with the bad names. Sofixed this one and the others...THANKS for catching this!

File: apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V90__new_schema_rename_to_original.sql (hunk 1 above)
Comment (mackec1) [reply apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V90__new_schema_rename_to_original.sql:96]: apparently my response didn't take. Yes, that is correct. During the table rename(s), any constraint will be correctly set to the new talbe name.

File: apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V90__new_schema_rename_to_original.sql [hunk 6]
```diff
@@ -0,0 +1,534 @@
...
+
+-- change the name of the index for bene_id
+alter index ${logic.psql-only} if exists 
+    public.medicare_beneficiaryid_history_new_bene_id_idx rename to medicare_beneficiaryid_history_bene_id_idx;
+
+alter table if exists public.medicare_beneficiaryid_history_new
+${logic.hsql-only} add constraint medicare_beneficiaryid_history_to_benficiaries foreign key (bene_id)
+${logic.hsql-only}     references public.beneficiaries_new (bene_id);
+${logic.psql-only} rename constraint medicare_beneficiaryid_history_new_to_benficiaries
+${logic.psql-only}     to medicare_beneficiaryid_history_to_benficiaries;
```
Comment (christrotteradhoc) [inline apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V90__new_schema_rename_to_original.sql:121]: typo in the new name (the old name typo seems legit but we can fix the new name here like you did for beneficiaries_history

File: apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V90__new_schema_rename_to_original.sql [hunk 7]
```diff
@@ -0,0 +1,534 @@
...
+
+alter index ${logic.psql-only} if exists 
+    public.beneficiary_monthly_new_partd_contract_year_month_bene_id_idx
+        rename to beneficiary_monthly_partd_contract_year_month_bene_id_idx;
+
+alter table if exists public.beneficiary_monthly_new
+${logic.hsql-only} add constraint beneficiary_monthly_to_benficiaries foreign key (bene_id)
+${logic.hsql-only}     references public.beneficiaries_new (bene_id);
+${logic.psql-only} rename constraint beneficiary_monthly_new_to_benficiaries
+${logic.psql-only}     to beneficiary_monthly_to_benficiaries;
```
Comment (christrotteradhoc) [inline apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V90__new_schema_rename_to_original.sql:144]: another case where we can fix the new name

File: apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V90__new_schema_rename_to_original.sql [hunk 8]
```diff
@@ -0,0 +1,534 @@
...
+${logic.hsql-only}     drop constraint public.medicare_beneficiaryid_history_new_pkey;
+${logic.hsql-only} alter table public.medicare_beneficiaryid_history_new
+${logic.hsql-only}     add constraint medicare_beneficiaryid_history_pkey primary key (bene_mbi_id);
+
+-- change the name of the index for bene_id
+alter index ${logic.psql-only} if exists 
+    public.medicare_beneficiaryid_history_new_bene_id_idx rename to medicare_beneficiaryid_history_bene_id_idx;
+
+alter table if exists public.medicare_beneficiaryid_history_new
+${logic.hsql-only} add constraint medicare_beneficiaryid_history_to_benficiaries foreign key (bene_id)
```
[the remark this answers is not in the document]
Comment (christrotteradhoc) [reply apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V90__new_schema_rename_to_original.sql:118]: V91

File: apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V90__new_schema_rename_to_original.sql (hunk 6 above)
Comment (christrotteradhoc) [reply apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V90__new_schema_rename_to_original.sql:121]: V91

File: apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V90__new_schema_rename_to_original.sql (hunk 7 above)
Comment (christrotteradhoc) [reply apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V90__new_schema_rename_to_original.sql:144]: V91

**Review by christrotteradhoc — APPROVED** [review]

**Pull request merged**